dialogs.c

```diff
@@ -585,6 +587,14 @@ static void refresh_usb(void)
 
 	/* Fill things in */
 	connect_clear(dialogs.connect_usb);
+	gdk_threads_leave();
```

Comment:
At line 570 there is a return. I believe gdk_threads_leave() should be called just before that return as well.

Reply:
I'll add this.